drivers/storage/portworx/component/securitycontextconstraints.go

```diff
@@ -0,0 +1,181 @@
+	// By design cross-namespace owner reference is invalid.
+	for _, scc := range s.getSCCs(cluster) {
+		err := s.k8sClient.Get(context.TODO(),
+			types.NamespacedName{
+				Name: scc.Name,
+			},
+			&ocp_secv1.SecurityContextConstraints{})
+
+		if errors.IsNotFound(err) {
+			err = s.k8sClient.Create(context.TODO(), &scc)
+		}
+
+		if err != nil {
+			return fmt.Errorf("failed to create %s security context constraints: %s", scc.Name, err)
+		}
+	}
+	return nil
+}
+
+func (s *scc) Delete(cluster *opcorev1.StorageCluster) error {
+	// Do not try to delete it if CRD does not exist, otherwise portworx would raise a cleanup failure warning.
+	if !s.crdExists(cluster) {
+		return nil
+	}
+
+	// Do not delete SCC during uninstallation, as it may be required by node wiper.
+	// It will be deleted when storage cluster is deleted as there is owner reference.
+	if cluster.DeletionTimestamp != nil &&
+		cluster.Spec.DeleteStrategy != nil &&
+		cluster.Spec.DeleteStrategy.Type != "" {
+		return nil
+	}
+
+	for _, scc := range s.getSCCs(cluster) {
+		err := s.k8sClient.Delete(context.TODO(), &scc)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *scc) MarkDeleted() {
+}
+
+func (s *scc) getSCCs(cluster *opcorev1.StorageCluster) []ocp_secv1.SecurityContextConstraints {
+	return []ocp_secv1.SecurityContextConstraints{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: PxSCCName,
+			},
+			AllowHostDirVolumePlugin: true,
+			AllowHostIPC:             true,
+			AllowHostNetwork:         true,
+			AllowHostPID:             true,
+			AllowHostPorts:           true,
+			AllowPrivilegeEscalation: boolPtr(true),
+			AllowPrivilegedContainer: true,
+			AllowedUnsafeSysctls:     []string{"*"},
+			AllowedCapabilities: []corev1.Capability{
+				ocp_secv1.AllowAllCapabilities,
+			},
+			FSGroup: ocp_secv1.FSGroupStrategyOptions{
+				Type: ocp_secv1.FSGroupStrategyRunAsAny,
+			},
+			Priority:               pointer.Int32Ptr(2),
+			ReadOnlyRootFilesystem: false,
+			RunAsUser: ocp_secv1.RunAsUserStrategyOptions{
+				Type: ocp_secv1.RunAsUserStrategyRunAsAny,
+			},
+			SELinuxContext: ocp_secv1.SELinuxContextStrategyOptions{
+				Type: ocp_secv1.SELinuxStrategyRunAsAny,
+			},
+			SeccompProfiles: []string{"*"},
+			SupplementalGroups: ocp_secv1.SupplementalGroupsStrategyOptions{
+				Type: ocp_secv1.SupplementalGroupsStrategyRunAsAny,
+			},
+			Volumes: []ocp_secv1.FSType{"*"},
+			Groups:  nil,
+			Users: []string{
+				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, pxutil.PortworxServiceAccountName(cluster)),
+				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, CSIServiceAccountName),
+				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, LhServiceAccountName),
+				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, PVCServiceAccountName),
+				fmt.Sprintf("system:serviceaccount:%s:%s", cluster.Namespace, CollectorServiceAccountName),
```

Comment:
Do we use privileged SCC for metrics collector?

Reply:
Yes.